### PR TITLE
Support legacy `- SeeAlso:` callout asides 

### PR DESCRIPTION
**Pull request by ethan-kusters**
Bug/issue #, if applicable: rdar://55141543

There’s a lot of existing Swift documentation content that uses
the legacy `- seealso:` style aside:

```md
- SeeAlso: This other content.
```

While Swift-DocC has always supported a number of these legacy asides
like:

```md
- Experiment: Lorem ipsum.
- Tip: Lorem ipsum

<!-- etc... -->
```

it doesn’t currently support `SeeAlso` which causes breakage
when migrating existing Swift doc comments over to Swift-DocC.

This resolves the issue.

## Dependencies

- https://github.com/apple/swift-markdown/pull/96

## Testing

Create documentation content with a SeeAlso aside and confirm it renders as expected:

```md
Hi there!

- SeeAlso: <https://swift.org/documentation/docc>
```

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by ethan-kusters** [comment]
@swift-ci please test

File: Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift [hunk 1]
```diff
@@ -295,11 +295,9 @@ public enum RenderBlockContent: Equatable {
     
     /// A type the describes an aside style.
     public struct AsideStyle: Codable, Equatable {
-        private static let specialDisplayNames: [String: String] = [
-            "nonmutatingvariant": "Non-Mutating Variant",
-            "mutatingvariant": "Mutating Variant",
-            "todo": "To Do",
-        ]
+        private static let knownDisplayNames: [String: String] = Dictionary(
+            uniqueKeysWithValues: Markdown.Aside.Kind.allCases.map { ($0.rawValue.lowercased(), $0.displayName) }
```
Comment (ethan-kusters) [inline Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift:299]: This moves ownership of the `displayName` over to Swift-Markdown where the existing ownership of asides is. Moving forward, we'll be able to add new supported asides with a single PR.

A bit of a discussion about this here: https://github.com/apple/swift-markdown/pull/96#discussion_r1065263050

**Review by QuietMisdreavus — APPROVED** [review]
Looks great, thanks!

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Pull request merged**